lab-sarah/ng-template/test/edit-gallery-controller-test.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+
+describe('testing edit-gallery controller', function(){
+
+  beforeEach(() => {
+    angular.mock.module('demoApp');
+    angular.mock.inject(($rootScope, $componentController, $httpBackend, authService) => {
+      authService.setToken('1234');
+
+      this.authService = authService;
+      this.$rootScope = $rootScope;
+      this.$componentController = $componentController;
+      this.$httpBackend = $httpBackend;
+    });
+  });
+
+  afterEach(() => {
+    this.authService.logout();
+  });
+
+  it('testing component bindings', () => {
+    let mockBindings = {
+      gallery: {
+        name: 'lulwat',
+        desc: 'my cool beach adventure',
+      },
+    };
+
+    let editGalleryCtrl = this.$componentController('editGallery', null, mockBindings);
+    expect(editGalleryCtrl.gallery.name).toEqual(mockBindings.gallery.name);
+    expect(editGalleryCtrl.gallery.desc).toEqual(mockBindings.gallery.desc);
+    this.$rootScope.$apply();
+  });
+
+  describe('testing #updateGallery', () => {
+
+    it('should make a valid PUT request', () => {
+      let url = 'http://localhost:3000/api/gallery/12345';
+      let headers = {
+        'Accept': 'application/json',
+        'Content-Type': 'application/json',
+        Authorization: 'Bearer 1234',
+      };
+
+      //update the name with new name instead of lulwat
+      this.$httpBackend.expectPUT(url, {_id: '12345', name: 'new name', desc: 'hello'}, headers)
+      .respond(200);
+
+      let mockBindings = {
+        gallery: {
+          _id: '12345',
+          name: 'lulwat',
+          desc: 'hello',
+        },
+      };
+
+      //editGalleryCtrl will be the gallery object and the updateGallery method
+      let editGalleryCtrl = this.$componentController('editGallery', null, mockBindings);
+      editGalleryCtrl.gallery.name = 'new name';
+
+      editGalleryCtrl.updateGallery();
```

Comment:
You are not testing anything related to your method here... only that your mock bindings were applied to your controller initially. To properly test the behavior of your method, you'll need to resolve something from your updateGallery method and then execute an expect statement within a then block chained off of it to test that your updateGallery is resolving as expected (and what is expected).